### PR TITLE
Class name fix e

### PR DESCRIPTION
**Pull request by kauket**

### CODE REVIEW

File: src/ReturnBookControl.java [hunk 1]
```diff
@@ -1,77 +1,75 @@
 class ReturnBookControl {
 
-	private ReturnBookUI ui;
-	private enum CONTROL_STATE { INITIALISED, READY, INSPECTING };
-	private CONTROL_STATE state;
+	private ReturnBookUi returnBookUi;
+	private enum controlState { INITIALISED, READY, INSPECTING };
```
Comment (johngalvin09) [inline src/ReturnBookControl.java]: Enum variables should start with a Capital letter

File: src/ReturnBookControl.java [hunk 2]
```diff
@@ -1,77 +1,75 @@
 class ReturnBookControl {
 
-	private ReturnBookUI ui;
-	private enum CONTROL_STATE { INITIALISED, READY, INSPECTING };
-	private CONTROL_STATE state;
+	private ReturnBookUi returnBookUi;
+	private enum controlState { INITIALISED, READY, INSPECTING };
+	private controlState state;
```
Comment (johngalvin09) [inline src/ReturnBookControl.java]: Unnecessary line breaks between variables

Comment (johngalvin09) [inline src/ReturnBookControl.java]: Unnecessary line breaks between variables

File: src/ReturnBookControl.java [hunk 3]
```diff
@@ -1,77 +1,75 @@
...
 	ReturnBookControl() {
 		this.library = library.INSTANCE();
-		state = CONTROL_STATE.INITIALISED;
+		state = controlState.INITIALISED;
 	}
 	
 	
-	void setUI(ReturnBookUI ui) {
-		if (!state.equals(CONTROL_STATE.INITIALISED)) {
+	void setUI(ReturnBookUi returnBookUi) {
```
Comment (johngalvin09) [inline src/ReturnBookControl.java]: should setUI be setUi to conform to the rest of the changes?

File: src/ReturnBookControl.java [hunk 4]
```diff
@@ -1,77 +1,75 @@
 class ReturnBookControl {
 
-	private ReturnBookUI ui;
-	private enum CONTROL_STATE { INITIALISED, READY, INSPECTING };
-	private CONTROL_STATE state;
+	private ReturnBookUi returnBookUi;
```
Comment (johngalvin09) [inline src/ReturnBookControl.java]: Tabs used instead of 4x spaces.

File: src/ReturnBookControl.java [hunk 5]
```diff
@@ -1,77 +1,75 @@
...
+		state = controlState.READY;
 	}
 
 
 	void bookScanned(int bookId) {
-		if (!state.equals(CONTROL_STATE.READY)) {
+		if (!state.equals(controlState.READY)) {
 			throw new RuntimeException("ReturnBookControl: cannot call bookScanned except in READY state");
 		}	
 		book currentBook = library.Book(bookId);
```
Comment (johngalvin09) [inline src/ReturnBookControl.java]: Unnecessary line break

File: src/ReturnBookControl.java [hunk 6]
```diff
@@ -1,77 +1,75 @@
...
 		double overDueFine = 0.0;
 		if (currentLoan.isOverDue()) {
 			overDueFine = library.calculateOverDueFine(currentLoan);
 		}
-		ui.display("Inspecting");
-		ui.display(currentBook.toString());
-		ui.display(currentLoan.toString());
+		returnBookUi.display("Inspecting");
+		returnBookUi.display(currentBook.toString());
+		returnBookUi.display(currentLoan.toString());
```
Comment (johngalvin09) [inline src/ReturnBookControl.java]: Unecessary line break on line 49

File: src/ReturnBookUI.java [hunk 7]
```diff
@@ -1,87 +1,77 @@
...
-	public void display(Object object) {
-		output(object);
-	}
-	
-	public void setState(UI_STATE state) {
-		this.state = state;
-	}
-
-	
+class ReturnBookUi {
```
Comment (johngalvin09) [inline src/ReturnBookUI.java]: class name changed but not file name of class. Not sure if this is an issue

File: src/ReturnBookUI.java [hunk 8]
```diff
@@ -1,87 +1,77 @@
...
+    private ReturnBookControl returnBookControl;
+    private Scanner input;
+    private uiState state;
+
+
+    ReturnBookUi(ReturnBookControl control) {
+        this.returnBookControl = control;
+        input = new Scanner(System.in);
+        state = uiState.INITIALISED;
+        control.setUI(this);
```
Comment (johngalvin09) [inline src/ReturnBookUI.java]: setUI should be setUi

File: src/ReturnBookControl.java (hunk 4 above)
Comment (kauket) [reply src/ReturnBookControl.java]: adjusted

File: src/ReturnBookControl.java (hunk 1 above)
Comment (kauket) [reply src/ReturnBookControl.java]: adjusted

File: src/ReturnBookControl.java (hunk 2 above)
Comment (kauket) [reply src/ReturnBookControl.java]: adjusted

File: src/ReturnBookControl.java (hunk 3 above)
Comment (kauket) [reply src/ReturnBookControl.java]: adjusted

File: src/ReturnBookControl.java (hunk 5 above)
Comment (kauket) [reply src/ReturnBookControl.java]: adjusted

File: src/ReturnBookControl.java (hunk 6 above)
Comment (kauket) [reply src/ReturnBookControl.java]: adjusted

File: src/ReturnBookUI.java [hunk 9]
```diff
@@ -1,87 +1,77 @@
...
-	}
-	
-	public void setState(UI_STATE state) {
-		this.state = state;
-	}
-
-	
+class ReturnBookUi {
+
+    public enum uiState {INITIALISED, READY, INSPECTING, COMPLETED}
```
Comment (johngalvin09) [inline src/ReturnBookUI.java]: Enum should start with capital

File: src/ReturnBookUI.java [hunk 10]
```diff
@@ -1,87 +1,77 @@
...
+
```
Comment (johngalvin09) [inline src/ReturnBookUI.java]: Unecessary line break on line 8

File: src/ReturnBookUI.java [hunk 11]
```diff
@@ -1,87 +1,77 @@
...
+            switch (state) {
+                case INITIALISED:
+                    break;
+                case READY:
+                    String bookString = input("Scan Book (<enter> completes): ");
+                    if (bookString.length() == 0) {
+                        returnBookControl.scanningComplete();
+                    } else {
+                        try {
+                            int bookId = Integer.valueOf(bookString).intValue();
```
Comment (johngalvin09) [inline src/ReturnBookUI.java]: can be Unboxed to: Integer.valueOf(bookString);

File: src/ReturnBookUI.java (hunk 9 above)
Comment (kauket) [reply src/ReturnBookUI.java]: adjusted

File: src/ReturnBookUI.java (hunk 8 above)
Comment (kauket) [reply src/ReturnBookUI.java]: adjusted

File: src/ReturnBookControl.java (hunk 2 above)
Comment (johngalvin09) [inline src/ReturnBookControl.java]: Sorry, still a break on line six that isn't required. - I am being pedantic but I think we need to be?

File: src/ReturnBookUI.java (hunk 10 above)
Comment (johngalvin09) [reply src/ReturnBookUI.java]: Sorry Line break on 7 should be deleted.

File: src/ReturnBookUI.java (hunk 11 above)
Comment (johngalvin09) [reply src/ReturnBookUI.java]: Ok cool cool, we can leave it 👍

**Comment by johngalvin09** [comment]
Chosen to leave the line breaks in the variables as it helps differentiate between ENUMS and other variables.

**Pull request merged**